cr/state/checkpoint_test.go

```diff
@@ -0,0 +1,113 @@
+package state
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/elastos/Elastos.ELA/common"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestKeyFrame_Deserialize(t *testing.T) {
+	frame := randomKeyFrame()
+
+	buf := new(bytes.Buffer)
+	frame.Serialize(buf)
+
+	frame2 := &KeyFrame{}
+	frame2.Deserialize(buf)
+
+	assert.True(t, keyFrameEqual(frame, frame2))
+}
+
+func TestKeyFrame_Snapshot(t *testing.T) {
+	frame := randomKeyFrame()
+	frame2 := frame.Snapshot()
+	assert.True(t, keyFrameEqual(frame, frame2))
+}
+
+func keyFrameEqual(first *KeyFrame, second *KeyFrame) bool {
+	for k := range first.Nicknames {
+		if _, ok := second.Nicknames[k]; !ok {
+			return false
+		}
+	}
+
```

Comment:
the length of first.CodeDIDMap and second.CodeDIDMap must be equal

Reply:
Nice catch, fixed